qdrant/deployment/src/main/java/io/quarkiverse/langchain4j/qdrant/QdrantDevServicesProcessor.java

```diff
@@ -0,0 +1,231 @@
+                    dockerStatusBuildItem,
+                    configuration,
+                    launchMode,
+                    devServicesConfig.timeout,
+                    !devServicesSharedNetworkBuildItem.isEmpty());
+
+            if (newDevService != null) {
+                devService = newDevService;
+                if (devService.isOwner()) {
+                    LOG.infof("Dev Services instance of Qdrant started (config: %s)", newDevService.getConfig());
+                }
+            }
+            if (devService == null) {
+                compressor.closeAndDumpCaptured();
+            } else {
+                compressor.close();
+            }
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        if (devService == null) {
+            return Collections.emptyList();
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = () -> {
+                shutdownContainers();
+                first = true;
+                cfg = null;
+            };
+            QuarkusClassLoader cl = (QuarkusClassLoader) Thread.currentThread().getContextClassLoader();
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(closeTask);
+        }
+        cfg = configuration;
+        result.add(devService.toBuildItem());
+        return result;
+    }
+
+    private void shutdownContainers() {
+        if (devService != null) {
+            try {
+                devService.close();
+            } catch (Throwable e) {
+                LOG.error("Failed to stop the Qdrant server", e);
+            } finally {
+                devService = null;
+            }
+        }
+    }
+
+    private DevServicesResultBuildItem.RunningDevService startQdrantContainer(
+            DockerStatusBuildItem dockerStatusBuildItem,
+            QdrantDevServiceCfg config,
+            LaunchModeBuildItem launchMode,
+            Optional<Duration> timeout,
+            boolean useSharedNetwork) {
+
+        if (!dockerStatusBuildItem.isDockerAvailable()) {
+            LOG.warn("Docker isn't working, please configure the Qdrant server location.");
+            return null;
+        }
+
+        QdrantContainer container = new QdrantContainer(
+                config.imageName(),
+                config.fixedPort(),
+                launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT ? config.serviceName() : null,
+                useSharedNetwork);
+
+        final Supplier<DevServicesResultBuildItem.RunningDevService> defaultQdrantSupplier = () -> {
+
+            // Starting Qdrant
+            timeout.ifPresent(container::withStartupTimeout);
+            container.start();
+
+            if (config.vectorCfg() != null) {
+                try (QdrantClient client = client(container)) {
+                    client
+                            .createCollectionAsync(
```

Comment:
I can only see the creation of collections here in the Dev Services code. Do we support automatically creating the collection when connecting to a non-devservices instance? (Most other embedding stores can do that) 
From looking at the code, the upstream implementation doesn't have that, so with this we would only get it for dev services, am I right?

Reply:
Ok, since we're also using the upstream client for now, I think this is something that should rather be implemented upstream, if at all. I agree it is not trivial to do this in a robust way.